tests/test_pipeline_mixin.py

```diff
@@ -236,67 +307,89 @@ def run_pipeline_test(
+                        f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: "
+                        f"Could not load the {key} from `{repo_id}` with `{name}`."
+                    )
+                    self.skipTest(f"Could not load the {key} from {repo_id} with {name}.")
+
+        # ---------------------------------------------------------
 
         # TODO: Maybe not upload such problematic tiny models to Hub.
-        if tokenizer is None and processor is None:
+        if tokenizer is None and "image_processor" not in processors and "feature_extractor" not in processors:
             logger.warning(
                 f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: Could not find or load "
-                f"any tokenizer / processor from `{repo_id}`."
+                f"any tokenizer / image processor / feature extractor from `{repo_id}`."
             )
             self.skipTest(f"Could not find or load any tokenizer / processor from {repo_id}.")
 
-        # TODO: We should check if a model file is on the Hub repo. instead.
-        try:
-            model = model_architecture.from_pretrained(repo_id, revision=commit)
-        except Exception:
-            logger.warning(
-                f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: Could not find or load "
-                f"the model from `{repo_id}` with `{model_architecture}`."
-            )
-            self.skipTest(f"Could not find or load the model from {repo_id} with {model_architecture}.")
-
         pipeline_test_class_name = pipeline_test_mapping[task]["test"].__name__
-        if self.is_pipeline_test_to_skip_more(pipeline_test_class_name, model.config, model, tokenizer, processor):
+        if self.is_pipeline_test_to_skip_more(pipeline_test_class_name, model.config, model, tokenizer, **processors):
             logger.warning(
                 f"{self.__class__.__name__}::test_pipeline_{task.replace('-', '_')}_{torch_dtype} is skipped: test is "
                 f"currently known to fail for: model `{model_architecture.__name__}` | tokenizer "
-                f"`{tokenizer_name}` | processor `{processor_name}`."
+                f"`{tokenizer_name}` | image processor `{image_processor_name}` | feature extractor `{feature_extractor_name}`."
             )
             self.skipTest(
-                f"Test is known to fail for: model `{model_architecture.__name__}` | tokenizer `{tokenizer_name}` | processor `{processor_name}`."
+                f"Test is known to fail for: model `{model_architecture.__name__}` | tokenizer `{tokenizer_name}` "
+                f"| image processor `{image_processor_name}` | feature extractor `{feature_extractor_name}`."
             )
 
         # validate
-        validate_test_components(self, task, model, tokenizer, processor)
+        validate_test_components(model, tokenizer)
```

Comment:
can you provide more details, please? this function is used only here and not using image_processor or feature_extractor (I updated the function signature accordingly)

Reply:
Yeah, it's more a design instead of what is actually happening so far.

The picture is that, a pipeline test is involved pipeline class (task), model, tokenizer/image processor/feature extractor/processor, and all of these **could** determine what we want to perform. (although currently only model and tokenizer appears in the body)

Reply:
Thanks for the clarification, I understand your point. 
While this is a design decision, I'm not entirely comfortable with it. The function isn't widely used or overridden in subclasses, it's only used once in the codebase. Not using arguments might cause confusion about what's actually happening in the function and could give the false impression that the image processor and tokenizer are being validated by it, but they are not. We can always add additional arguments to the function if they become necessary.

Reply:
works for me.

(FYI: unused arguments exist in `class Pipeline.__init__` and I agree with your above point)